snuba/utils/streams/consumer.py

```diff
@@ -97,6 +97,37 @@ def unsubscribe(self) -> None:
     def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
         raise NotImplementedError
 
+    @abstractmethod
+    def pause(self, partitions: Sequence[Partition]) -> None:
+        """
+        Pause consuming from the provided partitions.
+
+        A partition that is paused will be automatically resumed during
+        reassignment. If partitions should remain paused across rebalances,
+        this should be implemented in the assignment callback.
+
+        If any of the provided partitions are not in the assignment set, an
+        exception will be raised and no partitions will be paused.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def resume(self, partitions: Sequence[Partition]) -> None:
+        """
+        Resume consuming from the provided partitions.
+
+        If any of the provided partitions are not in the assignment set, an
+        exception will be raised and no partitions will be resumed.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def paused(self) -> Sequence[Partition]:
```

Comment:
I want this to return a `Set` instead of a `Sequence` but I don't want to make that change consistent everywhere, so leaving it this way for now.